app-android/tutashared/src/main/java/de/tutao/tutashared/alarms/AlarmModel.kt

```diff
@@ -82,15 +83,22 @@ object AlarmModel {
 
 			// This map + filter prevent an infinity loop trap by removing invalid rules like POS 320 for freq. weekly
 			// and ensures that 0 <= abs(SETPOS) < eventCount
+			var areAllSetPosValid = true
 			val parsedSetPos = setPosRules.map {
-				if (it.interval.toInt() < 0) {
-					expandedEvents.count() - abs(it.interval.toInt())
-				} else {
-					it.interval.toInt() - 1
+				try {
```

Comment:
similar to the other case and we could do it with nullable again but we could do it even simpler and just put try/catch around the whole `map()` and return, like

```kotlin
val parsedSetPos = try {
  setPosRules.map {
    // ...
  }.filter {
    // ...
  }
} catch (e: NumberFormatException) {
    Log.w()
    break
}

you should also tell what happened in the log message, it's not clear from the context what is failed